Error `ECONNREFUSED` from certain external links

I'm getting `ECONNREFUSED` from certain links. These errors don't appear consistently all the time, but they do appear more often. Here're some examples of those links:

```
   7:22   error  https://tools.ietf.org/html/rfc6749 is dead. (request to https://tools.ietf.org/html/rfc6749 failed, reason: connect ECONNREFUSED 64.170.98.42:443)                  no-dead-link
  16:140  error  https://tools.ietf.org/html/rfc6749#page-10 is dead. (request to https://tools.ietf.org/html/rfc6749#page-10 failed, reason: connect ECONNREFUSED 64.170.98.42:443)  no-dead-link
   7:451  error    https://www.slf4j.org/ is dead. (request to https://www.slf4j.org/ failed, reason: connect ECONNREFUSED 83.173.251.158:443)  no-dead-link
   35:116  error    http://www.slf4j.org/manual.html is dead. (request to http://www.slf4j.org/manual.html failed, reason: connect ECONNREFUSED 83.173.251.158:80)  no-dead-link
    7:115  error    https://logback.qos.ch/ is dead. (request to https://logback.qos.ch/ failed, reason: connect ECONNREFUSED 83.173.251.158:443)                                                    no-dead-link
  110:84   error    https://logback.qos.ch/manual/configuration.html is dead. (request to https://logback.qos.ch/manual/configuration.html failed, reason: connect ECONNREFUSED 83.173.251.158:443)  no-dead-link
  151:201  error    http://lucene.apache.org/solr/ is dead. (request to http://lucene.apache.org/solr/ failed, reason: connect ECONNREFUSED 95.216.24.32:80)  no-dead-link

```

I'm using version `4.4.3` of this linter rule. I have been trying to debug this issue but haven't found any pattern or cause of this yet.

--- a/src/no-dead-link.js
+++ b/src/no-dead-link.js
@@ -20,6 +20,7 @@ const DEFAULT_OPTIONS = {
   interval: 500, // The length of time in milliseconds before the interval count resets. Must be finite. [Experimental]
   intervalCap: 8, // The max number of runs in the given interval of time. [Experimental]
   keepAlive: false, // {boolean} if it is true, use keepAlive for checking request [Experimental]
+  userAgent: 'textlint-rule-no-dead-link/1.0' // {String} a UserAgent
 };
 
 // Adopted from http://stackoverflow.com/a/3809435/951517
@@ -120,7 +121,7 @@ const createFetchWithRuleDefaults = (ruleOptions) => {
       // to avoid ECONNRESET error
       // https://github.com/textlint-rule/textlint-rule-no-dead-link/issues/111
       headers: {
-        'User-Agent': 'textlint-rule-no-dead-link/1.0',
+        'User-Agent': ruleOptions.userAgent,
         'Accept': '*/*',
         // Same host for target url
         // https://github.com/textlint-rule/textlint-rule-no-dead-link/issues/111

--- a/test/no-dead-link.js
+++ b/test/no-dead-link.js
@@ -106,6 +106,13 @@ tester.run('no-dead-link', rule, {
         ignoreRedirects: true,
       },
     },
+    // https://github.com/textlint-rule/textlint-rule-no-dead-link/issues/128
+    {
+      text: 'should treat 200 OK. It requires browser-like User-Agent: https://issues.jenkins-ci.org/browse/JENKINS-59261',
+      options: {
+        userAgent: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.106 Safari/537.36'
+      }
+    }
   ],
   invalid: [
     {

